src/operator.rs

```diff
@@ -109,6 +109,27 @@ impl<'mir, 'tcx> EvalContextExt<'tcx> for super::MiriEvalContext<'mir, 'tcx> {
                     err!(InvalidPointerMath)
                 }
             }
+            Gt | Ge if left.is_ptr() && right.is_bits() => {
```

Comment:
I mean it doesn't really matter with int2ptr around the corner, but why is this not implemented commutatively?

Reply:
I didn't find a good way to do this without copying the code and, as you said, with int2ptr around the corner it just does not seem worth the effort.